lib/services/database.go

```diff
@@ -390,6 +390,58 @@ func IsRDSClusterSupported(cluster *rds.DBCluster) bool {
 	return true
 }
 
+// IsRDSInstanceAvailable checks if the RDS instance is available.
+func IsRDSInstanceAvailable(instance *rds.DBInstance) bool {
+	// For a full list of status values, see:
+	// https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/accessing-monitoring.html
+	switch aws.StringValue(instance.DBInstanceStatus) {
+
+	// Statues marked as "Not billed" in the above guide.
+	case "creating", "deleting", "failed",
+		"inaccessible-encryption-credentials", "incompatible-network",
+		"incompatible-restore":
+		return false
+
+	// Statuses marked as "Billed for storage" in the above guide.
+	case "inaccessible-encryption-credentials-recoverable", "starting",
+		"stopped", "stopping":
+		return false
+	}
+
+	return true
```

Comment:
In the future, when AWS adds more status types, we are likely to miss one which isn't actually "available." I wonder which default is better for unrecognized status codes: available or not?

In any case, it may be worthwhile to explicitly handle currently known "available" statuses and issue a warning on encountering unrecognized ones.

This question is applicable to other new functions.

Reply:
for unknown statuses, the default behaviour should be "available" to be more permissive. I think thats why i am doing blacklist. 

Worst case scenario an unknown-unavailable status made it through, user will see it in `tsh db ls`, then `tsh db connect` fails, which is what users see today.

I think it's better than the opposite, that the users will miss their databases for unknown-available statues.

Also, many of the statuses are **"transient"** like modifying configs so we don't want to flip the databases back and forth.  

---

I think explicitly handling "available" statuses is a good idea and I will add them here.

As for logging a warning on unknown statuses, not sure if that provides a lot of values to the customers when they see it as there's not much they can do with it. How about a debug message? WDYT?

Reply:
> How about a debug message? WDYT?

I was thinking about logging such states, too, though I was leaning toward warning as debug is not enabled by default and switching it on in production feels like a heavy hammer to use. Given that these warnings would only be logged in response to user interaction, I don't think there is a risk of drowning the log.